src/Symfony/Component/Security/Http/HttpUtils.php

```diff
@@ -135,6 +135,10 @@ public function generateUri($request, $path)
             return $path;
         }
 
+        if (!$request instanceof Request) {
```

Comment:
But we never add any similar checks if the doctype restricts the argument to a particular type.